src/ip-addresses.ts

```diff
@@ -45,27 +47,37 @@ const getPrivateIpAddress = (ipAddresses: IpAddresses) => {
   return ipAddresses.private;
 };
 
-export function parseIpAddresses(
-  ipResponse: sqladmin_v1beta4.Schema$IpMapping[] | undefined
-): IpAddresses {
-  if (!ipResponse) {
+const getPSCIpAddress = (ipAddresses: IpAddresses) => {
+  if (!ipAddresses.psc) {
     throw new CloudSQLConnectorError({
-      message: 'Cannot connect to instance, it has no supported IP addresses',
-      code: 'ENOSQLADMINIPADDRESS',
+      message: 'Cannot connect to instance, PSC address not found',
+      code: 'ENOPSCSQLADMINIPADDRESS',
     });
   }
+  return ipAddresses.psc;
+};
 
+export function parseIpAddresses(
+  ipResponse: sqladmin_v1beta4.Schema$IpMapping[] | undefined,
+  dnsName: string | null | undefined
```

Comment:
if `PSC` is not enabled on an instance, does the node client return `null` for `dnsName`? I know the Go client defaults it to empty string.

Reply:
Good question, I'm following the type definitions from [`@googleapis/sqladmin`](https://github.com/googleapis/google-api-nodejs-client/blob/main/src/apis/sqladmin/v1beta4.ts#L427) which sounds like the safest approach here.

Reply:
Sounds good to me! 👍